metaflow/plugins/airflow/airflow_utils.py

```diff
@@ -0,0 +1,410 @@
+                    data_dict[k] = parse_args(v)
+                elif isinstance(v, datetime):
+                    data_dict[k] = v.isoformat()
+                elif isinstance(v, timedelta):
+                    data_dict[k] = dict(seconds=v.total_seconds())
+                else:
+                    data_dict[k] = v
+            return data_dict
+
+        return parse_args(self._args)
+
+    @classmethod
+    def deserialize(cls, data_dict):
+        def parse_args(dd, type_check_dict):
+            kwrgs = {}
+            for k, v in dd.items():
+                if k not in type_check_dict:
+                    kwrgs[k] = v
+                    continue
+                if isinstance(v, dict) and isinstance(type_check_dict[k], dict):
+                    kwrgs[k] = parse_args(v, type_check_dict[k])
+                elif type_check_dict[k] == datetime:
+                    kwrgs[k] = datetime.fromisoformat(v)
+                elif type_check_dict[k] == timedelta:
+                    kwrgs[k] = timedelta(**v)
+                else:
+                    kwrgs[k] = v
+            return kwrgs
+
+        return cls(**parse_args(data_dict, cls._arg_types))
+
+
+def _kubernetes_pod_operator_args(operator_args):
+    from kubernetes import client
+
+    from airflow.kubernetes.secret import Secret
+
+    # Set dynamic env variables like run-id, task-id etc from here.
+    secrets = [
+        Secret("env", secret, secret) for secret in operator_args.get("secrets", [])
+    ]
+    args = operator_args
+    args.update(
+        {
+            "secrets": secrets,
+            # Question for (savin):
+            # Default timeout in airflow is 120. I can remove `startup_timeout_seconds` for now. how should we expose it to the user?
+        }
+    )
+    # Below cannot be passed in dictionary form. After trying a few times it didin't work.
+    additional_env_vars = [
+        client.V1EnvVar(
+            name=k,
+            value_from=client.V1EnvVarSource(
+                field_ref=client.V1ObjectFieldSelector(field_path=str(v))
+            ),
+        )
+        for k, v in {
+            "METAFLOW_KUBERNETES_POD_NAMESPACE": "metadata.namespace",
+            "METAFLOW_KUBERNETES_POD_NAME": "metadata.name",
+            "METAFLOW_KUBERNETES_POD_ID": "metadata.uid",
+            "METAFLOW_KUBERNETES_SERVICE_ACCOUNT_NAME": "spec.serviceAccountName",
+        }.items()
+    ]
+    args["env_vars"] = [
+        client.V1EnvVar(name=x["name"], value=x["value"]) for x in args["env_vars"]
+    ] + additional_env_vars
+
+    # We need to explicitly parse resources to k8s.V1ResourceRequirements otherwise airflow tries
+    # to parse dictionaries to `airflow.providers.cncf.kubernetes.backcompat.pod.Resources` object via
+    # `airflow.providers.cncf.kubernetes.backcompat.backward_compat_converts.convert_resources` function
+    resources = args.get("resources")
+    args["resources"] = client.V1ResourceRequirements(
```

Comment:
GPUs?

Reply:
GPU's are included in the compilation of resources in the `Airflow`(airflow.py) object